packages/amplify-provider-awscloudformation/src/root-stack-builder/root-stack-builder.ts

```diff
@@ -0,0 +1,252 @@
+  private _rootStackOptions: RootStackOptions; // options to help generate  cfn template
+  private _command: CommandType;
+  constructor(options: RootStackTransformOptions, command: CommandType) {
+    this._resourceConfig = options.resourceConfig;
+    this._command = command;
+  }
+
+  public async transform(): Promise<Template> {
+    // parse Input data
+    this._rootStackOptions = await this.getInput(); // get RootStackOPtions from cli-inputs.json
+    // generate cfn stack
+    const template = await this.generateRootStackTemplate();
+    // save stack
+    if (this._command === CommandType.PUSH) {
+      await this.deployOverrideStacksToDisk({
+        templateStack: template,
+      });
+    }
+    return template;
+  }
+  /**
+   *
+   * @returns Object required to generate Stack using cdk
+   */
+  private getInput = async (): Promise<RootStackOptions> => {
+    if (this._command === CommandType.INIT) {
+      const buildConfig: RootStackOptions = {
+        event: this._command,
+        rootStackFileName: this._resourceConfig.stackFileName,
+      };
+      return buildConfig;
+    } else {
+      const projectPath = pathManager.findProjectRoot();
+      const buildConfig: RootStackOptions = {
+        event: this._command,
+        rootStackFileName: this._resourceConfig.stackFileName,
+        overrideDir: pathManager.getOverrideDirPath(projectPath, 'root'),
+      };
+      return buildConfig;
+    }
+  };
+
+  /**
+   * Generates Root stack Template
+   * @returns CFN Template
+   */
+  private generateRootStackTemplate = async (): Promise<Template> => {
+    const stack = new AmplifyRootStack(undefined as any, 'Amplify');
+    const stackOutputs = new AmplifyRootStackOutputs(undefined as any, 'AmplifyOutputs');
+    // merge the outputs in one to get the combined object
+    Object.assign(stack.templateObj, stackOutputs.templateObj);
+    if (this._rootStackOptions.event === CommandType.INIT) {
+      // no override required
+      //apply init modifiers if any
+    }
+
+    if (this._rootStackOptions.event === CommandType.PUSH) {
+      // apply override here during push
+      //applyoverrides(stack.templateObj);
+      const projectPath = pathManager.findProjectRoot();
+      const overridePath = pathManager.getOverrideDirPath(projectPath, 'root');
+      const { overrideProps } = await import(path.join(overridePath, 'build', 'override.js'));
+      stack.templateObj = overrideProps(stack.templateObj);
+    }
+
+    const cfnRootStack: Template = stack.toCloudFormation();
+    const cfnRootStackOutputs: Template = stackOutputs.toCloudFormation();
+    Object.assign(cfnRootStack.Outputs, cfnRootStackOutputs.Outputs);
+    return cfnRootStack;
+  };
+
+  private deployOverrideStacksToDisk = async (props: DeploymentOptions) => {
+    const rootFilePath = path.join(pathManager.getBackendDirPath(), 'awscloudformation', 'build', this._resourceConfig.stackFileName);
```

Comment:
the full path could come from `pathManager` or at least the constants like build and the provider name.